pkg/storageos/scheduler_extender.go

```diff
@@ -0,0 +1,280 @@
+
+	// Create the deployment.
+	schedulerDeployment := s.schedulerDeployment(replicas)
+	return s.createOrUpdateObject(schedulerDeployment)
+}
+
+// schedulerDeployment returns a scheduler extender Deployment object. This
+// contains the deployment configuration of the external kube-scheduler.
+func (s Deployment) schedulerDeployment(replicas int32) *appsv1.Deployment {
+	podLabels := podLabelsForScheduler(s.stos.Name)
+	dep := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      schedulerExtenderName,
+			Namespace: s.stos.Spec.GetResourceNS(),
+			Labels: map[string]string{
+				"app": "storageos",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: podLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: podLabels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: SchedulerSA,
+					Containers:         s.schedulerContainers(),
+					Volumes:            s.schedulerVolumes(),
+				},
+			},
+		},
+	}
+
+	return dep
+}
+
+// schedulerContainers returns a list of containers that should be part of the
+// scheduler extender deployment.
+func (s Deployment) schedulerContainers() []corev1.Container {
+	return []corev1.Container{
+		{
+			Image:           s.stos.Spec.GetHyperkubeImage(s.k8sVersion),
+			Name:            "storageos-scheduler",
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Args: []string{
+				"kube-scheduler",
+				"--config=/storageos-scheduler/config.yaml",
+				"-v=4",
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "storageos-scheduler-config",
+					MountPath: "/storageos-scheduler",
+				},
+			},
+		},
+	}
+}
+
+// schedulerVolumes returns a list of volumes that should be part of the
+// scheduler extender deployment.
+func (s Deployment) schedulerVolumes() []corev1.Volume {
+	return []corev1.Volume{
+		{
+			Name: "storageos-scheduler-config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: "storageos-scheduler-config"},
+				},
+			},
+		},
+	}
+}
+
+// deleteSchedulerExtender deletes all the scheduler related resources.
+func (s Deployment) deleteSchedulerExtender() error {
+	if err := s.deleteObject(s.getDeploymentByName(schedulerExtenderName)); err != nil {
+		return err
+	}
+	if err := s.deleteObject(s.getConfigMap(policyConfigMapName)); err != nil {
+		return err
+	}
+	if err := s.deleteObject(s.getConfigMap(schedulerConfigConfigMapName)); err != nil {
+		return err
+	}
+	if err := s.deleteClusterRoleBinding(SchedulerClusterBindingName); err != nil {
+		return err
+	}
+	if err := s.deleteServiceAccount(SchedulerSA); err != nil {
+		return err
+	}
+	if err := s.deleteClusterRole(SchedulerClusterRoleName); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Deleting all the related resources here to avoid cluttering `Delete()`.
Will change the same for CSI helpers in a separate PR.